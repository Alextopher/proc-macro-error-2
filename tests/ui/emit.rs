extern crate test_crate;
use test_crate::*;

emit!(one, two, three, four);

fn main() {}
