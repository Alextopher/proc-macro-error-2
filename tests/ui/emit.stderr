warning: panic message is not a string literal
   --> $DIR/lib.rs:472:12
    |
472 |     panic!(AbortNow)
    |            ^^^^^^^^
    |
    = note: `#[warn(non_fmt_panic)]` on by default
    = note: this is no longer accepted in Rust 2021
help: add a "{}" format string to Display the message
    |
472 |     panic!("{}", AbortNow)
    |            ^^^^^
help: or use std::panic::panic_any instead
    |
472 |     std::panic::panic_any(AbortNow)
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: 1 warning emitted

warning: panic message is not a string literal
   --> $DIR/lib.rs:472:12
    |
472 |     panic!(AbortNow)
    |            ^^^^^^^^
    |
    = note: `#[warn(non_fmt_panic)]` on by default
    = note: this is no longer accepted in Rust 2021
help: add a "{}" format string to Display the message
    |
472 |     panic!("{}", AbortNow)
    |            ^^^^^
help: or use std::panic::panic_any instead
    |
472 |     std::panic::panic_any(AbortNow)
    |     ^^^^^^^^^^^^^^^^^^^^^^

warning: 1 warning emitted

error: emit!(span, from) test
 --> $DIR/emit.rs:4:7
  |
4 | emit!(one, two, three, four, five);
  |       ^^^

error: emit!(span, expr1, expr2) test
 --> $DIR/emit.rs:4:12
  |
4 | emit!(one, two, three, four, five);
  |            ^^^

error: emit!(span, single_expr) test
 --> $DIR/emit.rs:4:17
  |
4 | emit!(one, two, three, four, five);
  |                 ^^^^^

error: Diagnostic::emit() test
 --> $DIR/emit.rs:4:24
  |
4 | emit!(one, two, three, four, five);
  |                        ^^^^

error: emit_call_site_error!(expr) test
 --> $DIR/emit.rs:4:1
  |
4 | emit!(one, two, three, four, five);
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: This is an error

  = note: simple note
  = help: simple help
  = help: simple hint
  = note: simple yay
  = note: format note
  = note: Some note
  = note: spanned simple note
  = note: spanned format note
  = note: Some note

 --> $DIR/emit.rs:5:13
  |
5 | emit_notes!(one, two);
  |             ^^^
